docs/basic-features/layouts.md

```diff
@@ -0,0 +1,109 @@
+---
+description: Learn how to share components and state between Next.js pages with Layouts.
+---
+
+# Layouts
+
+<details open>
+  <summary><b>Examples</b></summary>
+  <ul>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/layout-component">layout-component</a></li>
+  </ul>
+</details>
+
+The React model allows us to deconstruct a [page](/docs/basic-features/pages.md) into a series of components. Many of these components are often reused between pages. For example, you might have the same navigation bar and footer on every page.
+
+```jsx
+// components/layout.js
+
+import Navbar from './navbar'
+import Footer from './footer'
+
+export default function Layout({ children }) {
+  return (
+    <>
+      <Navbar />
+      <main>{children}</main>
+      <Footer />
+    </>
+  )
+}
+```
+
+## Examples
+
+### Single Shared Layout with Custom App
+
+If you only have one layout for your entire application, you can create a [Custom App](/docs/advanced-features/custom-app.md) and wrap your application with the layout. Since the `<Layout />` component is re-used when changing pages, its component state will be preserved (e.g. input values).
+
+```jsx
+// pages/_app.js
+
+import Layout from '../components/layout'
+
+export default function MyApp({ Component, pageProps }) {
+  return (
+    <Layout>
+      <Component {...pageProps} />
+    </Layout>
+  )
+}
+```
+
+### Per-Page Layouts
+
+If you need multiple layouts, you can add a property `getLayout` to your page, allowing you to return a React component for the layout. This allows you to define the layout on a _per-page basis_. Since we're returning a function, we can have complex nested layouts if desired.
+
+```jsx
+// pages/index.js
+
+import Layout from '../components/layout'
+import NestedLayout from '../components/nested-layout'
+
+export default function Page() {
+  return {
+    /** Your content */
+  }
+}
+
+Page.getLayout = (page) => (
+  <Layout>
+    <NestedLayout>{page}</NestedLayout>
+  </Layout>
+)
```

Comment:
Is it possible to for the layout to fetch data? E.g. if it renders a navigation with articles or similar. Might be worth documenting that too since it's a quite common use case

Reply:
Yes, it is!